http3/headers.go

```diff
@@ -164,3 +164,23 @@ func hostnameFromRequest(req *http.Request) string {
 	}
 	return ""
 }
+
+func responseFromHeaders(headerFields []qpack.HeaderField) (*http.Response, error) {
+	hdr, err := parseHeaders(headerFields, false)
+	if err != nil {
+		return nil, err
+	}
+	rsp := &http.Response{
+		Proto:         "HTTP/3.0",
+		ProtoMajor:    3,
+		Header:        hdr.Headers,
+		ContentLength: hdr.ContentLength,
+	}
+	status, err := strconv.Atoi(hdr.Status)
+	if err != nil {
```

Comment:
Should we check that the resulting integer status is a valid HTTP status so that we reject values like 12345?

Reply:
There's some considerations on this in https://datatracker.ietf.org/doc/html/rfc9110#name-status-codes. Looks like there are reasons to not reject all values outside of the valid range.